spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/config/features/SchedulerConfiguration.java

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright 2018 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.dataflow.server.config.features;
+
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.boot.autoconfigure.condition.AllNestedConditions;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.boot.autoconfigure.jdbc.DataSourceProperties;
+import org.springframework.boot.context.properties.EnableConfigurationProperties;
+import org.springframework.cloud.dataflow.configuration.metadata.ApplicationConfigurationMetadataResolver;
+import org.springframework.cloud.dataflow.registry.AppRegistryCommon;
+import org.springframework.cloud.dataflow.server.config.apps.CommonApplicationProperties;
+import org.springframework.cloud.dataflow.server.repository.TaskDefinitionRepository;
+import org.springframework.cloud.dataflow.server.service.SchedulerService;
+import org.springframework.cloud.dataflow.server.service.impl.DefaultSchedulerService;
+import org.springframework.cloud.dataflow.server.service.impl.TaskConfigurationProperties;
+import org.springframework.cloud.scheduler.spi.core.Scheduler;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Conditional;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.context.annotation.Import;
+import org.springframework.core.io.ResourceLoader;
+import org.springframework.transaction.annotation.EnableTransactionManagement;
+
+/**
+ * Establishes the {@link SchedulerService} instance to be used by SCDF.
+ *
+ * @author Glenn Renfro
+ */
+
+@Configuration
+@Conditional({SchedulerConfiguration.SchedulerConfigurationPropertyChecker.class})
+@EnableConfigurationProperties({TaskConfigurationProperties.class, CommonApplicationProperties.class})
+@Import(TaskConfiguration.class)
+@EnableTransactionManagement
```

Comment:
Actually, one of my earlier comments was to remove this `@Import(TaskConfiguration.class)` here as it is already imported in `FeaturesConfiguration`.

Reply:
Yeah.  I wouldn't expect us to need to do `@Import` in most cases since the server is just a Boot app so it should be picked up via classpath scanning, right?

Reply:
We don't do the classpath scanning instead via `DataFlowServerAutoConfiguration` which imports the `DataFlowServerConfiguration` and then we conditionally import these configuration classes (e.g TaskConfiguration when task feature is enabled).  Hence, importing the `TaskConfiguration` here again in the `SchedulerConfiguration` may not be needed.